src/BlockTypes/CollectionFilters.php

```diff
@@ -184,18 +183,111 @@ private function get_aggregated_collection_data( $block ) {
 	/**
 	 * Get all inner blocks recursively.
 	 *
-	 * @param WP_Block $block The block to get inner blocks from.
-	 * @param array    $results The results array.
+	 * @param WP_Block_List $inner_blocks The block to get inner blocks from.
+	 * @param array         $results      The results array.
 	 *
 	 * @return array
 	 */
-	private function get_inner_blocks_recursive( $block, $results = array() ) {
-		$results[] = $block->name;
-		if ( ! empty( $block->inner_blocks ) ) {
-			foreach ( $block->inner_blocks as $inner_block ) {
-				$results = $this->get_inner_blocks_recursive( $inner_block, $results );
+	private function get_inner_blocks_recursive( $inner_blocks, &$results = array() ) {
+		if ( is_a( $inner_blocks, 'WP_Block_List' ) ) {
+			foreach ( $inner_blocks as $inner_block ) {
+				$results[] = $inner_block->name;
+				$this->get_inner_blocks_recursive(
+					$inner_block->inner_blocks,
+					$results
+				);
 			}
 		}
+
 		return $results;
 	}
```

Comment:
This function can be refactored into this:
```
	private function get_inner_blocks_recursive( $inner_blocks, &$results = array() ) {
		if ( ! is_array( $inner_blocks ) ) {
			$inner_blocks = iterator_to_array( $inner_blocks );
		}

		return array_reduce(
			$inner_blocks,
			function( $acc, $inner_block ) {
				$acc[] = $inner_block->name;
				return $this->get_inner_blocks_recursive(
					$inner_block->inner_blocks,
					$acc
				);
			},
			$results
		);
	}
```

Reply:
Do you think this looks better? 🤔 I kinda like the first version with the loop to be honest.

Reply:
Without the short circuit, I do, but `iterator_to_array()` only accepts arrays from PHP 8.2, so we need that short circuit, which kinda decreases the readability.

We can remove the short circuit to make the logic flow less divergent, but I don't really like the inline condition.

```
	private function get_inner_blocks_recursive( $inner_blocks, &$results = array() ) {
		return array_reduce(
			is_a( $inner_blocks, 'WP_Block_List' ) ? iterator_to_array( $inner_blocks ) : array(),
			function( $acc, $inner_block ) {
				$acc[] = $inner_block->name;
				return $this->get_inner_blocks_recursive(
					$inner_block->inner_blocks,
					$acc
				);
			},
			$results
		);
	}
```